### PR TITLE
STYLE: Enable unnecessary list index lookup

### PR DESCRIPTION
**Pull request by uzzell**
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

Associated with #48855.  Enables the Pylint type "R" warning `unnecessary-list-index-lookup`.  Also updates the list of intentionally disabled checks.

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -4368,7 +4368,7 @@ def write_data(self, chunksize: int | None, dropna: bool = False) -> None:
         bvalues = []
         for i, v in enumerate(values):
             new_shape = (nrows,) + self.dtype[names[nindexes + i]].shape
-            bvalues.append(values[i].reshape(new_shape))
+            bvalues.append(v.reshape(new_shape))
```
Comment (MarcoGorelli) [inline pandas/io/pytables.py:4371]: nice!

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for working on this

could you please:
- run the pre-commit checks (see contributing guide for how)
- fetch upstream, and rebase onto it, looks like you have some irrelevant changes in here

**Comment by uzzell** [comment]
@MarcoGorelli , are you talking about the changes to `pyproject.toml`?  I took the liberty of updating the list of intentionally disabled warnings: `wrong-import-order` appeared twice in the list, while `broad-except` and `overridden-final-method` were in the list at the top of #48855 but not in the section for intentionally disabled warnings in `pyproject.toml`.

If you'd like, I can change them back.

**Comment by MarcoGorelli** [comment]
ah, that's fine then, thanks for explaining!

**Review by MarcoGorelli — APPROVED** [review]
thanks @uzzell

**Pull request merged**